src/fetcher.js

```diff
@@ -1110,8 +1113,42 @@ class Fetcher {
     options.method = method
     options.body = options.data || options.body
     options.force = true
+    const fetcher = this
 
-    return this.fetch(uri, options)
+    if (options.body && !options.contentType) {
+      throw new Error('Web operation sending data must have a defined contentType.')
+    }
+    if (options.contentType) {
+      options.headers = options.headers || {}
+      options.headers['content-type'] = options.contentType
+    }
+    if (Fetcher.withCredentials(uri, options)) {
+      options.credentials = 'include' // Otherwise coookies are not sent
+    }
+
+    return new Promise(function (resolve, reject) {
+      fetcher._fetch(uri, options).then(response => {
+        if (response.ok) {
+          if (response.body) {
+            response.text().then(data => {
+              response.responseText = data
+              resolve(response)
+            })
+          } else {
+            resolve(response)
+          }
+        } else {
+          let msg = 'Web error: ' + response.status
+          if (response.statusText) msg += ' (' + response.statusText + ')'
+          msg += ' on ' + method + ' of <' + uri + '>'
+          if (response.responseText) msg += ': ' + response.responseText
+          reject(new Error(msg))
+        }
+      }, err => {
+        let msg = 'Fetch error for ' + method + ' of <' + uri + '>:' + err
```

Comment:
Not sure if the suggested class above should handle this as well, but having a reference to the original error object is useful for further processing. The `toString` function could be updated to return this message if an exception rather than a response was given.